src/annoylib.h

```diff
@@ -84,6 +85,10 @@ using std::string;
 using std::pair;
 using std::numeric_limits;
 using std::make_pair;
+using std::mutex;
+
+mutex _node_mutex;
+mutex _roots_mutex;
```

Comment:
no reason to make these global

Reply:
I was trying to put them in the class with everything else, but I kept getting 'deleted function' errors when compiling.

Reply:
The exact warning (when compiling the precision_test.cpp example in Linux with the s_compile_cpp.sh script):

`error: use of deleted function ‘std::mutex::mutex(const std::mutex&)`

(similar warnings with MSVC).

Reply:
strange! i can fix later